src/main/java/de/cronn/testutils/h2/H2Util.java

```diff
@@ -174,7 +178,9 @@ private static long selectCount(Connection connection, String tableName) throws
 		try (ResultSet tables = con.getMetaData().getTables(null, null, null, tableTypes)) {
 			while (tables.next()) {
 				String tableName = tables.getString("TABLE_NAME");
-				tableNames.add(tableName);
+				if (!persistentTables.contains(tableName)) {
```

Comment:
Would be pretty mean if `getTablesNames()` would silently hide a table that is specific to some framework ;)
Please note that a library like this shouldn’t be tied to some implementation detail of another framework.

I have a suspicion of what you actually try to solve but maybe you can elaborate on it?
If my suspicion is right, I think there’s a better way to do it…